rnaseq-cgl-pipeline/wrapper.py

```diff
@@ -1,102 +1,203 @@
+    # If no arguments provided, print full help menu
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+    # Get name of most recent running container. If socket is mounted, should be this one.
+    try:
+        name = subprocess.check_output(['docker', 'ps', '--format', '{{.Names}}']).split('\n')[0]
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError('No container detected, ensure Docker is being run with: '
+                           '"-v /var/run/docker.sock:/var/run/docker.sock" as an argument. \n\n{}'.format(e.message))
     # Get name of mounted volume
     blob = json.loads(subprocess.check_output(['docker', 'inspect', name]))
     mounts = blob[0]['Mounts']
     # Ensure docker.sock is mounted correctly
     sock_mount = [x['Source'] == x['Destination'] for x in mounts if 'docker.sock' in x['Source']]
-    if len(sock_mount) != 1:
-        raise IllegalArgumentException('Missing socket mount. Requires the following:'
-                                       'docker run -v /var/run/docker.sock:/var/run/docker.sock')
+    require(len(sock_mount) == 1, 'Missing socket mount. Requires the following: '
+                                  'docker run -v /var/run/docker.sock:/var/run/docker.sock')
     # Ensure formatting of command for 2 mount points
     if len(mounts) == 2:
-        if not all(x['Source'] == x['Destination'] for x in mounts):
-            raise IllegalArgumentException('Docker Src/Dst mount points, invoked with the -v argument,'
-                                           'must be the same if only using one mount point aside from the '
-                                           'docker socket.')
+        require(all(x['Source'] == x['Destination'] for x in mounts),
+                'Docker Src/Dst mount points, invoked with the -v argument, '
+                'must be the same if only using one mount point aside from the docker socket.')
         work_mount = [x['Source'] for x in mounts if 'docker.sock' not in x['Source']]
     else:
         # Ensure only one mirror mount exists aside from docker.sock
         mirror_mounts = [x['Source'] for x in mounts if x['Source'] == x['Destination']]
         work_mount = [x for x in mirror_mounts if 'docker.sock' not in x]
-        if len(work_mount) > 1:
-            raise IllegalArgumentException('Too many mirror mount points provided, see documentation.')
-        if len(work_mount) == 0:
-            raise IllegalArgumentException('No required mirror mount point provided, see documentation.')
-    # Enforce file input standards
+        require(len(work_mount) == 1, 'Wrong number of mirror mounts provided, see documentation.')
+    # Look for inputs in work directory
+    star = glob(os.path.join(work_mount[0], 'star*'))
+    rsem = glob(os.path.join(work_mount[0], 'rsem*'))
+    kallisto = glob(os.path.join(work_mount[0], 'kallisto*'))
+    if not args.star:
```

Comment:
Wait. All this logic here is weird. Am I correct in understanding that if the user does _not_ pass a default path for the STAR/Kallisto/RSEM inputs, you are globbing in the work directory to find the inputs? Why are you doing that? It seems like the type of logic that might lead to someone inadvertently running the pipeline on the wrong index build, or something like that. I don't see much of a benefit to doing this, but let me know if I am missing some context.